### DISCUSSION TITLE
테스트 메서드명에 테스트하는 메서드가 드러나면 좋겠습니다.

### DISCUSSION THREAD

**Discussion by amaran-th** [BDD-CLUB/01-doo-re-back#42] [논의 사항]
가령 createToken()이라는 메서드를 테스트하는 메서드의 경우, 
```java
@Test
@DisplayName("[성공] 회원의 아이디를 입력하면 토큰을 생성한다.")
void createToken_success() {
```
이렇게 하면 DisplayName으로 상세한 테스트 정보를 나타내면서 메서드명으로는 어떤 메서드를 테스트하는지에 대한 정보를 함께 줄 수 있어서 테스트를 알아보기 더 쉬워질 것 같습니다.
이미 백엔드 컨벤션으로 정해졌던 내용이긴 하지만, 다시 논의하고 싶어 디스커션 올립니다!

**Comment by shkisme** [comment]
테스트 코드 역시 코드이기에, 읽는 사람을 위해 테스트 메서드 명으로 추가 정보를 나타낸다면 더 풍부한 표현이 가능해질 것 같습니다. 👍

**Reply by shkisme** [reply]
`@DisplayName`에 prefix로 `[성공]` `[실패]`를 두는 것 역시도 테스트를 한눈에 파악하기에 용이하다고 생각합니다! 마찬가지로 컨벤션으로 두면 좋겠습니다 😀

**Comment by lcqff** [comment]
그러면 실패 테스트의 경우 
`createToekn_실패사유_실패()`가 되는 건가요?

**Reply by lcqff** [reply]
아 상세 내용은 디스플레이 네임에 작성하고 메서드 명은 간단하게 `메서드명_success()`가 되는 건가요?
실패사유가 여러개인 경우 실패 테스트명이 겹칠 거 같은데,,,

**Reply by shkisme** [reply]
그러네요! 실패 사유가 여러개일 경우 메서드 명이 겹칠 경우를 방지하기 위해서 실패 사유는 추가로 작성하는 게 좋을 것 같은데 실패 사유를 영어로 표기하면 한영 변환 리소스가 생길 거 같네요 ㅠㅠ 메서드 명만 영어로 표기하는 건 어떤가요?

createToken(메서드명)_성공()
createToken(메서드명)_사유1_실패()
createToken(메서드명)_사유2_실패()

**Comment by JJimini** [comment]
네 좋은 것 같아요. 그런데 [성공]을 앞에 넣을 지 뒤에 넣을 지도 정하면 좋을 것 같아요

**Comment by lcqff** [comment]
현재 방식에서 메서드명 붙이기
ex) findById_삭제된_팀원은_조회되지_않는다_성공()
ex) update_팀의_정보는_변경된다_성공()
